src/rgw/rgw_sal_motr.cc

```diff
@@ -3493,7 +3493,30 @@ int MotrStore::list_users(const DoutPrefixProvider* dpp, const std::string& meta
                         std::string& marker, int max_entries, void *&handle,
                         bool* truncated, std::list<std::string>& users)
 {
-    return 0;
+  int rc;
+  bool max_entires_specified = true;
+  if(max_entries <= 0) {
+    max_entries = 1000; 
+    max_entires_specified = false;
+  }
+  vector<string> keys(max_entries + 1);
+  vector<bufferlist> vals(max_entries + 1);
+
+  keys[0] = marker;
+  rc = next_query_by_name(RGW_MOTR_USERS_IDX_NAME, keys, vals);
+  if (rc < 0) {
+    ldpp_dout(dpp, 0) << "ERROR: NEXT query failed. " << rc << dendl;
+    return rc;
+  }
+  if(!(keys.back()).empty() && max_entires_specified){
```

Comment:
it will evaluate nullptr to false 👍 
```suggestion
  if(keys.back() and max_entires_specified){
```

Reply:
> it will evaluate nullptr to false 

It won't because we are not getting null strings, we are getting empty strings. If there are 2 users and `max-entries` is 4, `next_query_by_name()` will return ["abc", "pqr", "", ""]

> `max_entries specified` is not needed as well I think.

It is required to set correct value of `truncated`

Reply:
> It won't because we are not getting null strings, we are getting empty strings. If there are 2 users and max-entries is 4, next_query_by_name() will return ["abc", "pqr", "", ""]

Ohh yes correct, it is a string type not bufferlist. Got it👍